### PR TITLE
[PRO-61] Add Interactive Menu

### PR DESCRIPTION
**Pull request by MatteoCultrera**
## Overview

The purpose of this MR is to allow a script to show some kind of menu to the user so that he could pick a selection.

In order to do so, we've added a new shell injectable class `InteractiveMenu`.
This class takes an array of options (`String`) and return the answer done by the user (as `String`). If there is an error, it returns `nil`.

## Description

In order to do so, we've created a new executable target. This target will start a new app that presents a single `SwiftUI` screen with some options.

The screens are inside yet another target, called `SwiftyScriptyAppViews`. This target is not exposed anywhere but just imported from `SwiftyScripty`. This is needed because if we imported it to the executable target, previews would not have worked.

The executable target is called `SwiftyScriptyApp`. This target has only a main that works as follow

- The main script runs a process where the `SwiftyScriptyApp` script is executed.
- The `SwiftyScriptyApp` script tries to read options to show to the user from the default pipe
- The `SwiftyScriptyApp` script shows the screen
- The user picks an option
- The `SwiftyScriptyApp` script writes the option to the pipe, and then kills himself
- The main script reads the pipe to get the option

This architecture is needed because if a process kills an app, it automatically terminates, so we need to instantiate a secondary process otherwise the main script would die after the selection

The two scripts use a common pipe to communicate. The data are written and read on the pipe use `Base64` as common encoding/decoding language.

## Example

[test.mov](https://github.com/user-attachments/assets/230346d0-da7c-41c8-a628-74885d7e8b32)

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -31,11 +35,28 @@ let package = Package(
         
         .target(
             name: "SwiftyScripty",
+            dependencies: ["SwiftyScriptyAppViews"],
             path: "Sources/SwiftyScripty",
             resources: [.copy("Resources")],
             swiftSettings: [.define("DEBUG", .when(configuration: .debug))]
         ),
 
+
```
Comment (danieleleto94) [inline Package.swift:44]:
```suggestion
```

File: Sources/SwiftyScripty/InteractiveMenu.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        for await data in process.stream {
+            switch data {
+            case .output(let string):
+                if
+                    let objectData = string.data(using: .utf8),
+                    let object = try? JSONDecoder().decode(SwiftyScripty.Menu.PassableData.self, from: objectData)
+                {
+                    return object
+                }
+            case .failureRunningProcess, .exitCode, .error:
```
Comment (danieleleto94) [inline Sources/SwiftyScripty/InteractiveMenu.swift:50]:
```suggestion
                }
                
            case .failureRunningProcess, .exitCode, .error:
```

File: Sources/SwiftyScriptyApp/InteractiveMenus/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+extension AppDelegate: PickerDelegate {
+    func didTap(option: SinglePickerOption) {
+        applicationDelegate.didSelect(
+            option: .init(representation: option.value, id: option.id)
+        )
+    }
+}
+
+class CustomWindow: NSWindow {
```
Comment (danieleleto94) [inline Sources/SwiftyScriptyApp/InteractiveMenus/AppDelegate.swift:60]: mettiamo in un altro file?

File: Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        self.delegate = delegate
+    }
+
+    public var body: some View {
+        VStack {
+            VStack(alignment: .leading) {
+                Text(title)
+                    .font(.largeTitle)
+                .padding(.vertical)
+                VStack(spacing: 10) {
```
Comment (danieleleto94) [inline Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift:41]:
```suggestion
                Text(title)
                    .font(.largeTitle)
                    .padding(.vertical)
                
                VStack(spacing: 10) {
```

File: Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+            .init(id: 3, value: "three"),
+            .init(id: 4, value: "four"),
+            .init(id: 5, value: "five"),
+            .init(id: 6, value: "six"),
+        ],
+        delegate: PickerDelegateMock()
+    )
+}
+
+class PickerDelegateMock: PickerDelegate {
```
Comment (danieleleto94) [inline Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift:86]: anche tutto quello che c'è qua sotto lo metterei in un altro file

File: Package.swift (hunk 1 above)
Comment (MatteoCultrera) [reply Package.swift]: fixed

File: Sources/SwiftyScripty/InteractiveMenu.swift (hunk 2 above)
Comment (MatteoCultrera) [reply Sources/SwiftyScripty/InteractiveMenu.swift]: fixed

File: Sources/SwiftyScriptyApp/InteractiveMenus/AppDelegate.swift (hunk 3 above)
Comment (MatteoCultrera) [reply Sources/SwiftyScriptyApp/InteractiveMenus/AppDelegate.swift]: si spostato

File: Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift (hunk 4 above)
Comment (MatteoCultrera) [reply Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift]: fixed

File: Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift (hunk 5 above)
Comment (MatteoCultrera) [reply Sources/SwiftyScriptyAppViews/Menus/SinglePicker.swift]: yess spostato

**Review by danieleleto94 — APPROVED** [review]

**Pull request merged**